src/syntax/read.rkt

```diff
@@ -165,6 +177,13 @@
         (for/list ([var arg-names] [repr var-reprs]) (cons var (representation-name repr)))
         '()))
 
+(define (verify-platform platform desired-platform)
+    (match platform
+      [`(platform ,platform-name)
+        (equal? platform-name desired-platform)]
+      [else
+        #f]))
```

Comment:
```
(equal? platform `(platform ,desired-platform)
```

Reply:
That said I think it would be better not to use `:description` (which per the spec should be a string) but instead use a custom property like `:herbie-platform` and then don't have the `(platform ...)` wrapper. Then the code would be simpler throughout.